### PR TITLE
test: add WebToolsHtmlExtractionTests and FallbackSearchProviderTests

### PR DESCRIPTION
**Pull request by cyberprophet**
## Summary
- Adds `WebToolsHtmlExtractionTests.cs` — 23 tests covering all private static HTML extraction helpers in `WebTools` via reflection: OG metadata (`og:title`, `og:description`, `og:image`), `<title>` tag, meta description, JSON-LD blocks (including truncation at 5000 chars), and `StripHtml` (script/style removal, entity decoding, newline collapsing, malformed input)
- Adds `FallbackSearchProviderTests.cs` — 13 tests covering the `FallbackSearchProvider` primary→secondary chain: primary success, primary throws / secondary succeeds, both throw (secondary exception propagates), `OperationCanceledException` not swallowed, default `numResults=8`, and result format pass-through
- All 85 existing + new tests pass; the 6 pre-existing `EmbeddedResourceTests` failures are unchanged

## Test plan
- [x] `dotnet test` from `agency.tests/` — 85 passed, 6 pre-existing failures (EmbeddedResourceTests), 0 skipped
- [x] No xUnit1051 warnings (all async calls use `TestContext.Current.CancellationToken`)
- [x] No source code modified — test-only PR

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by MintSurf — APPROVED** [review]
## Review Summary
- Outcome: approve
- Summary: The added coverage aligns with the current `FallbackSearchProvider` and `WebTools` behavior, and I did not find a correctness risk in the new tests. Local project checks could not be executed here because `dotnet` is not installed in this environment.

File: agency.tests/WebToolsHtmlExtractionTests.cs [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+
+/// <summary>
+/// Unit tests for the private static HTML extraction helpers in <see cref="WebTools"/>.
+/// Methods are invoked via reflection because they are private implementation details
+/// of the sealed class — source code must not be modified.
+/// </summary>
+public class WebToolsHtmlExtractionTests
+{
+    // Reflected method handles — resolved once per test class lifetime.
+    static readonly MethodInfo ExtractMetaField =
```
Comment (cyberchacha) [inline agency.tests/WebToolsHtmlExtractionTests.cs:13]: **<sub><sub>P3 Badge</sub></sub>  Reflection handles resolved at class level**

The static readonly fields with `?? throw` are a clean pattern — tests fail fast with a clear `MissingMethodException` if a method is renamed. Good defensive coding for reflection-based testing.

One minor note: if any of these methods are ever renamed in the source, all tests in this class will fail at construction time (not individually), which could make debugging slightly harder. Consider adding a brief XML comment on each field noting which source line it maps to, so future maintainers can quickly reconcile.

Useful? React with 👍 / 👎.

File: agency.tests/WebToolsHtmlExtractionTests.cs [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public void ExtractJsonLdText_EmptyHtml_ReturnsNull()
+    {
+        var result = InvokeExtractJsonLdText(string.Empty);
+
+        Assert.Null(result);
+    }
```
Comment (cyberchacha) [inline agency.tests/WebToolsHtmlExtractionTests.cs:310]: **<sub><sub>P3 Badge</sub></sub>  Truncation boundary assertion is precise**

Good catch testing `result!.Length <= 5003` (5000 chars + `"..."`). This exactly matches the implementation at `WebTools.cs:317-318`. The `Assert.EndsWith("...", result)` paired with the length check provides strong coverage of the truncation logic.

Useful? React with 👍 / 👎.

File: agency.tests/FallbackSearchProviderTests.cs [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        _secondary.SearchAsync(Arg.Any<string>(), Arg.Any<int>(), Arg.Any<CancellationToken>())
+                  .ThrowsAsync(new HttpRequestException("secondary unreachable"));
+
+        var ex = await Assert.ThrowsAsync<HttpRequestException>(() =>
+            _sut.SearchAsync("query", ct: TestContext.Current.CancellationToken));
+
+        Assert.Contains("secondary unreachable", ex.Message);
+    }
+
+    // ─── Cancellation behavior ────────────────────────────────────────────────
```
Comment (cyberchacha) [inline agency.tests/FallbackSearchProviderTests.cs:133]: **<sub><sub>P3 Badge</sub></sub>  Cancellation test correctly validates `OperationCanceledException` propagation**

This test correctly verifies that `OperationCanceledException` is re-thrown (not swallowed as a fallback trigger), matching the `catch (OperationCanceledException) { throw; }` in `FallbackSearchProvider.cs:18`. The companion test at line 148 confirming secondary is never called completes the coverage nicely.

Useful? React with 👍 / 👎.

**Review by cyberchacha — APPROVED** [review]
## Review Summary
- Outcome: approve
- Summary: Clean, test-only PR adding 36 well-structured unit tests for `WebTools` HTML extraction helpers and `FallbackSearchProvider` fallback chain. Tests correctly exercise all code paths including edge cases (empty input, HTML encoding, truncation, cancellation propagation). Reflection-based approach for private methods is a reasonable tradeoff documented in code comments.

> Note: `dotnet build`/`dotnet test` could not run due to missing NuGet restore and a missing project reference (`Agency/Agency.csproj`). Review is based on static analysis of the diff against the source implementation.
